src/zenml/stack/stack_component.py

```diff
@@ -33,21 +33,7 @@
 logger = get_logger(__name__)
 
 
-class StackComponent(BaseModel, ABC):
-    """Abstract StackComponent class for all components of a ZenML stack.
-
-    Attributes:
-        name: The name of the component.
-        uuid: Unique identifier of the component.
-    """
-
-    name: str
-    uuid: UUID = Field(default_factory=uuid4)
-
-    # Class Configuration
-    TYPE: ClassVar[StackComponentType]
-    FLAVOR: ClassVar[str]
-
+class StackComponentConfig(BaseModel, ABC):
```

Comment:
I haven't looked into this PR in detail, but I'd be interested how you solved that mypy correctly recognizes the type of `self.config` in all the various stack component subclasses?

Reply:
We haven't yet. Any suggestions how to do that cleanly?